app/services/nomis/client.rb

```diff
@@ -18,11 +18,20 @@ def initialize(root)
       end
     end
 
+    # rubocop:disable Metrics/MethodLength
     def get(route, queryparams: {}, extra_headers: {})
       response = request(
         :get, route, queryparams: queryparams, extra_headers: extra_headers
       )
-      data = JSON.parse(response.body)
+
+      # Elite2 can return a 204 to mean empty results, and we don't know if
+      # it is meant to be a {} or a []. For now, we are going to use nil and
+      # let the caller handle it.
+      if response.status == 204
+        data = nil
```

Comment:
Wonder if it makes life easier just returning [] so you don't need to modify callers

Reply:
That's what I did originally, but we don't know if it's a [] or {} :(